cpu/esp32/Makefile.dep

```diff
@@ -6,10 +6,6 @@ USEMODULE += esp_idf_driver
 USEMODULE += esp_idf_esp32
 USEMODULE += esp_idf_soc
 
-ifneq (,$(filter newlib,$(USEMODULE)))
-  DEFAULT_MODULE += newlib_nano
-endif
```

Comment:
Is `newlib_nano` no longer supported?

Reply:
gcc 8.4.0 for xtensa uses nanaospecs but doesn't have a separate include dir `newlib-nano`. Therefore the test in `makefiles/libc/newlib.mk` for this directory fails and the compilation aborts. Therefore `newlib-nano` can't be used anymore if we don't allow that this directory doesn't exist.

Reply:
```
makefiles/libc/newlib.mk:98: *** USE_NEWLIB_NANO==1 but nano include folder not found!
```

Reply:
I already thought about to change `newlib.mk`.https://github.com/RIOT-OS/RIOT/blob/2520b5c608d393947b76b3ca8329b558c1c59dc6/makefiles/libc/newlib.mk#L97-L99
This check was introduced with PR #11145 to avoid `-isystem -I...` if the directory is not found. This assumes that a toolchain providing `newlib-nano` has such a separate include directory for the nano version of `newlib.h`. This is at least true for ARM and RISC-V.

Although `crosstool-NG` for ESP32 generates both `libc.a` and `libc_nano.a` and uses the `nano.specs`, it does not generate separate versions for `newlib.h`. The reason is probably that `crosstool-NG` for ESP32 is maintained by the developers of the `ESP-IDF` SDK which only uses the nano version of the `newlib`.

A possible solution would be to make the setting of `-isystem` dependent on whether the `newlib-nano` directory was found. This would require that a toolchain which only provides the nano version of the `newlib` is accepted.
```diff
-  ifeq (,$(NEWLIB_NANO_INCLUDE_DIR))
-    $(error USE_NEWLIB_NANO==1 but nano include folder not found!)
+  ifneq (,$(NEWLIB_NANO_INCLUDE_DIR))
+     # newlib-nano overrides newlib.h and its include dir should therefore go before
+     # the regular system include dirs.
+     INCLUDES := -isystem $(NEWLIB_NANO_INCLUDE_DIR) $(INCLUDES)
   endif
-
-  # newlib-nano overrides newlib.h and its include dir should therefore go before
-  # the regular system include dirs.
-  INCLUDES := -isystem $(NEWLIB_NANO_INCLUDE_DIR) $(INCLUDES)
 endif
```

Reply:
How about

```patch
--- a/makefiles/libc/newlib.mk
+++ b/makefiles/libc/newlib.mk
@@ -94,7 +94,7 @@ ifeq (1,$(USE_NEWLIB_NANO))
                                                     $(NEWLIB_INCLUDE_DIR)/newlib/nano \
                                                     $(NEWLIB_INCLUDE_DIR)/nano))
 
-  ifeq (,$(NEWLIB_NANO_INCLUDE_DIR))
+  ifeq (,$(NEWLIB_NANO_INCLUDE_DIR) $(NEWLIB_NANO_NO_INCLUDE_DIR))
     $(error USE_NEWLIB_NANO==1 but nano include folder not found!)
   endif
```

and setting `NEWLIB_NANO_NO_INCLUDE_DIR` to 1 for esp32?

Or does it not make a differnce and newlib-nano is always used either way?

Reply:
This wouldn't be correct, because the line following 
https://github.com/RIOT-OS/RIOT/blob/2520b5c608d393947b76b3ca8329b558c1c59dc6/makefiles/libc/newlib.mk#L103
would then still try to add `-isystem $(NEWLIB_NANO_INCLUDE_DIR)` which isn't defined and would lead exactly to that problem the check should prevent.

For ESP32 we have only two options, either we compile the toolchain without `newlib_nano` and the standard version is always used or we compile the toolchain with `newlib_nano`, but then it has to be used always because only be one version of the `newlib.h` can exist.

I have provided my proposal as PR #17553, let's see what @kaspar30 says.

Reply:
Ok, then please squash

Reply:
I PR #17553 is OK, I will drop commit e7caa97. It might make sense to wait for it before squashing.